### DISCUSSION TITLE
Since chess uci is no longer supported, what is a simple fix for older code that uses this module?

### DISCUSSION THREAD

**Discussion by infinitless** [niklasf/python-chess#848] [Q&A] [answered]
Question says it all. Was trying to run this https://github.com/mptedesco/python-chess-analysis but then ran into chess.uci not existing. 

Is there an easy fix? Also notified the author so any comments might help them too!

**Comment by niklasf** [comment, marked as answer]
The new `chess.engine` module can do everything that `chess.uci` could and vice versa, but there is no mechanical recipe to convert from `chess.uci` to `chess.engine`. However, usually the process is straight-foward. In this case:

1. `chess.uci.popen_engine(path)` becomes `chess.engine.SimpleEngine.popen_uci(path)`, if you do not intend to use `asyncio`.
2. `InfoHandler` has been removed. No need to register one, anymore.
3. `engine.position(board)` and `engine.go(time=time_in_millis)` for the purpose of analysis is now a single call `info = engine.analyse(board, chess.engine.Limit(time=time_in_seconds))`.
4. `info["pv"][0]` is the best move according to the engine.
5. `info["score"]` will be a [`PovScore`](https://python-chess.readthedocs.io/en/latest/engine.html#chess.engine.PovScore), that you can use to get the centi-pawn or mate evaluation from the desired perspective.

**Reply by TomHagenWills** [reply]
@infinitless did you manage to have Matthew P. Tedesco´s script work? I have tried to adapt it, but I am beginner in Python. I would greatly appreciate your help!

**Reply by infinitless** [reply]
I am a beginner too, unfortunately. It all depends on what kind of application you are writing!

**Reply by TomHagenWills** [reply]
Thanks for taking the time to answer. I am  trying to develop a script to analyze my games, identify good/bad moves, visualize evaluations, etc.
There seems to be no comprehensive documentation of the python chess package, unfortunately.
FYI I found an interesting script "Centipawn Loss Calculator" in the description of this video:
https://www.youtube.com/watch?v=YnnJ0Da4Rp0&t=406s

**Reply by infinitless** [reply]
Actually, I would encourage you to join Discord and look for communities there - there are literally hundreds of people developing Python Chess apps and they talk regularly on Discord. I had most of my queries answered here by some really good and talented people! https://discord.com/invite/python
